### PR TITLE
security image bug

### PR DESCRIPTION
**Pull request by davidkelly**
When switching encryption algorithms (initial one was not available in linux), I forgot to increase the size of the key vector (to 32 bytes, so I now use a 32-byte hash -- Sha256), so decryption failed consistently.  Also, when looking for leaks, I deleted the decrypted buffer before returning it (doh).

### CODE REVIEW

**Comment by davidkelly** [comment]
Test Plan

Start fresh (no hifikey or hifi_image)
Create wallet, selecting a security image
Quit interface
Restart interface
Open Wallet
Same security image should appear
Change the image
Quit interface
Restart interface
Go to wallet
The new image should appear
:beers:

File: interface/src/commerce/Wallet.cpp [hunk 1]
```diff
@@ -331,7 +331,6 @@ bool Wallet::decryptFile(const QString& inputFilePath, unsigned char** outputBuf
     *outputBufferSize += tempSize;
     *outputBufferPtr = outputBuffer;
     qCDebug(commerce) << "decrypted buffer size" << *outputBufferSize;
-    delete[] outputBuffer;
```
Comment (howard-stearns) [inline interface/src/commerce/Wallet.cpp]: so does the client of outputBufferPtr delete it?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2797/HighFidelity-Beta-PR11246-c316e50f0f74a24271922946dad07d4d90ebc37b.dmg

File: interface/src/commerce/Wallet.cpp (hunk 1 above)
Comment (davidkelly) [reply interface/src/commerce/Wallet.cpp]: Yes -- though I see it isn't doing so now, will push that in a sec.

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2808/HighFidelity-Beta-PR11246-0f722992a5d2a078c01c6a81d76c711fd313208b.exe

**Pull request merged**